src/main/java/DAO/DBConnection.java

```diff
@@ -0,0 +1,42 @@
+package dao;
+
+import dao.exceptions.DAOException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DbConnection {
+    private static DbConnection instance;
+    private static DbAccess access;
+
+    public DbConnection() {
+        getInstance();
+    }
+
+    public DbConnection(String properties) {
+        getInstance(properties);
+    }
+
+    private static synchronized DbConnection getInstance() throws DAOException {
+        if (instance == null) {
+            FileReader fileReader = FileReader.getInstance();
+            access = fileReader.getAccess("database.properties");
+            instance = new DbConnection();
+        }
+        return instance;
+    }
```

Comment:
Exception in thread "main" java.lang.StackOverflowError
	at java.base/sun.nio.ch.FileChannelImpl.endBlocking(FileChannelImpl.java:168)
	at java.base/sun.nio.ch.FileChannelImpl.readInternal(FileChannelImpl.java:820)
	at java.base/sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:797)
	at java.base/java.nio.file.FileChannelLinesSpliterator$1.read(FileChannelLinesSpliterator.java:140)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:287)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:188)
	at java.base/java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:329)
	at java.base/java.io.BufferedReader.readLine(BufferedReader.java:396)
	at java.base/java.nio.file.FileChannelLinesSpliterator.readLine(FileChannelLinesSpliterator.java:174)
	at java.base/java.nio.file.FileChannelLinesSpliterator.forEachRemaining(FileChannelLinesSpliterator.java:116)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at dao.FileReader.getData(FileReader.java:62)

Reply:
Do you have the same issue?